src/utils/hooks/api.js

```diff
@@ -16,14 +16,25 @@ const getErrorMessage = (response, type = 'q') => {
   return errorDictionary.getUnknownError;
 };
 
-export const useLunaticFetcher = () => {
+export const useGetReferentiel = (nomenclatures) => {
   const oidcClient = useContext(AuthContext);
+  const { apiUrl } = useContext(AppContext);
 
-  const lunaticFetcher = useConstCallback((url, options) =>
-    getFetcherForLunatic(oidcClient.accessToken)(url, options)
-  );
+  const getReferentiel = useConstCallback((refName) => {
+    const finalUrl = `${apiUrl}/api/nomenclature/${refName}`;
+    return getFetcherForLunatic(oidcClient.accessToken)(finalUrl);
+  });
 
-  return { lunaticFetcher };
+  const getReferentielForVizu = useConstCallback((refName) => {
+    if (nomenclatures && Object.keys(nomenclatures).includes(refName)) {
+      const finalUrl = nomenclatures[refName];
+      return getFetcherForLunatic(oidcClient.accessToken)(finalUrl);
+    }
+    // No nomenclature, return empty array to lunatic
+    return new Promise((resolve) => resolve([]));
```

Comment:
When a Promise needs to only "resolve" or "reject", it’s more efficient and readable to use the methods specially created for such use cases: Promise.resolve(value) and Promise.reject(error). (sonarlint)